src/generic/grid.cpp

```diff
@@ -11337,4 +11400,617 @@ wxGetContentRect(wxSize contentSize,
+        numberRows++;
+    if (grid->m_rowLabelWidth)
+        numberCols++;
+
+    *childCount = numberRows * numberCols;
+
+    return wxACC_OK;
+}
+
+// Gets the specified child (starting from 1).
+// If *child is null and return value is wxACC_OK, this means that the child
+// is a simple element and not an accessible object.
+wxAccStatus wxGridAccessible::GetChild(int childId, wxAccessible** child)
+{
+    wxGrid* grid = wxDynamicCast(GetWindow(), wxGrid);
+    wxCHECK( grid, wxACC_FAIL );
+
+    if ( childId == wxACC_SELF )
+    {
+        *child = this;
+    }
+    else
+    {
+        if ( !m_gridCellAccessible )
+        {
+            m_gridCellAccessible.reset(new wxGridCellAccessible(grid, childId));
+        }
+        else if ( m_gridCellAccessible->m_childId != childId )
+        {
+            // pass on the information whether the cell is on the same row or
+            // column than the previous one to avoid redundant coordinate info
+            wxGridCellCoords previous_coords = m_gridCellAccessible->m_coords;
+            m_gridCellAccessible.reset(new wxGridCellAccessible(grid, childId));
+            if ( m_gridCellAccessible->m_coords.GetRow() == previous_coords.GetRow() )
+            {
+                m_gridCellAccessible->m_isSameRow = true;
+            }
+            if ( m_gridCellAccessible->m_coords.GetCol() == previous_coords.GetCol() )
+            {
+                m_gridCellAccessible->m_isSameCol = true;
+            }
+        }
+        *child = m_gridCellAccessible.get();
+    }
+    return wxACC_OK;
+}
+
+// Returns a role constant.
+wxAccStatus wxGridAccessible::GetRole(int childId, wxAccRole* role)
+{
+    wxGrid* grid = wxDynamicCast(GetWindow(), wxGrid);
+    wxCHECK( grid, wxACC_FAIL );
+
+    if ( childId == wxACC_SELF )
+    {
+        *role = wxROLE_SYSTEM_TABLE;
+    }
+    else
+    {
+        // children can be cells or headers
+        wxGridCellCoords coords = ConvertChildIdToCoords(grid, childId);
+        if ( coords.GetRow() == -1 )
+            *role = wxROLE_SYSTEM_COLUMNHEADER;
+        else if ( coords.GetCol() == -1 )
+            *role = wxROLE_SYSTEM_ROWHEADER;
+        else
+            *role = wxROLE_SYSTEM_CELL;
+    }
+
+    return wxACC_OK;
+}
+
+// Returns a state constant.
+wxAccStatus wxGridAccessible::GetState(int childId, long* state)
+{
+    wxGrid* grid = wxDynamicCast(GetWindow(), wxGrid);
+    wxCHECK( grid, wxACC_FAIL );
+
+    if ( childId > 0 )
+        return wxACC_NOT_IMPLEMENTED;
+
+    long st = 0;
+    if ( !grid->IsEnabled() )
+        st |= wxACC_STATE_SYSTEM_UNAVAILABLE;
+    if ( !grid->IsShown() )
+        st |= wxACC_STATE_SYSTEM_INVISIBLE;
+
+    if( grid->IsFocusable() )
+        st |= wxACC_STATE_SYSTEM_FOCUSABLE;
+
+    // this needs to be returned such that a child below is recognized as focused
+    // (check for cursorRow and cursorCol being != -1?)
+    if ( grid->HasFocus() )
+        st |=  wxACC_STATE_SYSTEM_FOCUSED;
```

Comment:
Totally nitpickish but there is an extra space here:
```suggestion
        st |= wxACC_STATE_SYSTEM_FOCUSED;
```